src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisExtIntegrationTest.php

```diff
@@ -283,13 +283,14 @@ public function testFromDsnWithMultipleHosts()
     public function testJsonError()
     {
         $redis = new \Redis();
-        $connection = Connection::fromDsn('redis://localhost/json-error', ['delete_after_ack' => true], $redis);
+        $connection = Connection::fromDsn('redis://localhost/messenger-json-error', ['delete_after_ack' => true], $redis);
         try {
             $connection->add("\xB1\x31", []);
         } catch (TransportException $e) {
         }
 
         $this->assertSame('Malformed UTF-8 characters, possibly incorrectly encoded', $e->getMessage());
+        $redis->del('messenger-json-error');
```

Comment:
I think we should move this above the assertion so that the cleanup also happens when the assertion is not fulfilled.

Reply:
That is a good idea. 

@alexandre-daubois Could you to this for `testGetAfterReject()`, `testLazy()` and `testGetNonBlocking()` too?

Sometimes you may move things around like @xabbuh suggest, other times you may have to do `finally`.

```php
try {
  // assert
} finally {
  $redis->del(...);
}
```

Reply:
Here you go :slightly_smiling_face: Also improved `testJsonError()` code flow